aeronautical/asterix/repetitive.ksy

```diff
@@ -0,0 +1,55 @@
+meta:
+  id: repetitive
+  license: GPL-3.0-only
+  endian: be
+  ks-version: 0.9
+
+doc: |
+  Implementantion of ASTERIX format.
+
+  From Eurocontrol Asterix web page at
+  https://www.eurocontrol.int/asterix
+
+  Repetitive Standard Field format.
+
+  The repetitive field format is a variable lenght format.
+
+  It uses three parameters:
+
+  item, which is just the item name, as I001/010 or I062/390. It is just
+  informative, and is not used during parsing, but permits to consult the item
+  name later in the target language.
+
+  flag: this is a bool parameter, which determines if there have to be parsed or
+  just skipped, mainly used by compound. In Catalog files, the flag is always true
+
+  f_size, which is the size of the repetitive parts.
+
+params:
+  - id: name
+    type: str
+  - id: flag
+    type: b1
```

Comment:
`b1` as a param type is weird, I think `bool` type is more recognizable.

Reply:
Do we really have `bool` type? If we do, why not to remove it because `b1` seems to be enough?

Reply:
User guide at 6.1. Basic data types, say type spec for booleans is b1.

Reply:
@KOLANICH, @LmpOSX:

Seems that the user guide is too long, we should apparently strive to make it less boring. But please read at least the following excerpt from https://doc.kaitai.io/user_guide.html#param-types:

> `type` specification in `params` definition is slightly different from normal `type` specification in `seq` or `instances`. It specifies pure concept of "type", as would be used to represent data in a programming language, without any serialization details (like endianness, sizes, conversions, encodings, etc). Thus, one can't use any extra keys to specify that, but instead one can use:
> 
> * no type to specify byte arrays
> * `str` to specify strings
> * `bool` to specify booleans
> * `struct` to allow arbitrary KaitaiStruct-compatible user types
> * `io` to specify KaitaiStream-compatible IO streams
> * `any` to allow any type (if target language supports that)


I completely agree with this: `b1` type means "one bit", which is more like a serialization detail than specifying a data type. It might not be clear to someone who has no experience with Kaitai Struct, that the possible values of `b1` are booleans `true` or `false`, not integers `0` or `1`. There's nothing we can do with it in `seq` fields and instances, but I think for params `bool` type should be used instead. It makes more sense: you're passing a boolean, not a "one bit of data". And it's legible for all programmers, not just for the handful of people that work with Kaitai Struct every day.

Additionaly, I'd prefer to ban `b[0-64]` in params `type` completely, because it's IMHO exclusively a serialization detail . It doesn't really say which type will be used in the target language (FYI the current KSC uses `uint64` type for all bit types - except for `b1` of course). On top of that, it suggests that for example one byte (`s1`) won't fit in `b6` param, which apparently does not apply.

Feel free to leave `b1` there if you insist, but I personally think it harms specification readability and intelligibility, which I perceive as one of the great things on Kaitai Struct.

Reply:
>Additionaly, I'd prefer to ban b[0-64] in params type completely,

I don't think so. In future we may want to support languages like verilog, for them `b1` or `b3` as a parameter perfectly make sense.